File: kreartif-api/models/komentarModel.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');
const { User } = require('./userModel');

const Komentar = sequelize.define('Komentar', {
  isi: { type: DataTypes.TEXT, allowNull: false },
  user_id: { type: DataTypes.INTEGER, allowNull: false },
  karya_id: { type: DataTypes.INTEGER, allowNull: false },
});

Komentar.belongsTo(User, { foreignKey: 'user_id' });

exports.Komentar = Komentar;

File: kreartif-api/models/komentarModel.py
from sqlalchemy import Column, DateTime, Integer, Text, ForeignKey, func
from sqlalchemy.orm import relationship
from database import Base

class Komentar(Base):
    __tablename__ = "komentars"

    id = Column(Integer, primary_key=True, index=True)
    isi = Column(Text, nullable=False)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    karya_id = Column(Integer, ForeignKey("karyas.id"), nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    # Relasi ke user dan karya
    user = relationship("User", back_populates="komentar_list")
    karya = relationship("Karya", back_populates="komentar_list")
